fix: server exception loop

slb1988 opened the pull request:
When NetworkManager.SceneManager.SynchronizeNetworkObjects got exception, there will always got exception in AddClient for add ConnectedClients dictionary multi times. Client and Server all get stuck.

<!-- Replace this block with what this PR does and why. Describe what you'd like reviewers to know, how you applied the engineering principles, and any interesting tradeoffs made. Delete bullet points below that don't apply, and update the changelog section as appropriate. -->

<!-- Add short version of the JIRA ticket to the PR title (e.g. "feat: new shiny feature [MTT-123]") -->

<!-- Add RFC link here if applicable. -->

## Changelog

- Added: The package whose Changelog should be added to should be in the header. Delete the changelog section entirely if it's not needed.
- Fixed: If you update multiple packages, create a new section with a new header for the other package. 
- Removed/Deprecated/Changed: Each bullet should be prefixed with Added, Fixed, Removed, Deprecated, or Changed to indicate where the entry should go.

## Testing and Documentation

- No tests have been added.
- Includes unit tests.
- Includes integration tests.
- No documentation changes or additions were necessary.
- Includes documentation for previously-undocumented public API entry points.
- Includes edits to existing public API documentation.

<!--  Uncomment and mark items off with a * if this PR deprecates any API:
### Deprecated API
- [ ] An `[Obsolete]` attribute was added along with a `(RemovedAfter yyyy-mm-dd)` entry.
- [ ] An [api updater] was added.
- [ ] Deprecation of the API is explained in the CHANGELOG.
- [ ] The users can understand why this API was removed and what they should use instead.
-->

slb1988 committed:
fix server exception loop

When NetworkManager.SceneManager.SynchronizeNetworkObjects got exception, there will always got exception in AddClient for add ConnectedClients dictionary multi times. Client and Server all get stuck.

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/not_signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.netcode.gameobjects?pullRequest=2652) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.netcode.gameobjects?pullRequest=2652) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.netcode.gameobjects?pullRequest=2652) it.</sub>

NoelStephensUnity commented:
Interesting... I think I see what you are talking about with that change.
Will look at it closer tomorrow....but thank you very much for the PR submission and contributing to help make NGO better!  
👍 💯

slb1988 committed:
fix duplicate GlobalObjectIdHash & not save problem

Duplicate GlobalObjectIdHash come out always, globalObjectId got null in some phase like PrefabImporterEditor::OnDestroy, another problem is GlobalObjectIdHash changed but prefab don't.

slb1988 committed:
Merge pull request #1 from slb1988/slb1988-patch-1

fix duplicate GlobalObjectIdHash & not save problem

EmandM commented on com.unity.netcode.gameobjects/Runtime/Core/NetworkObject.cs:
```diff
@@ -59,8 +59,20 @@ internal void GenerateGlobalObjectIdHash()
             }
 
-            var globalObjectIdString = UnityEditor.GlobalObjectId.GetGlobalObjectIdSlow(this).ToString();
+            var globalObjectId = UnityEditor.GlobalObjectId.GetGlobalObjectIdSlow(this);
```
Do you mind explaining what this change is doing?

EmandM reviewed (commented):
Hi @slb1988! Our team are currently making a push to address open pull requests. If this is still something you're interested in landing, the NGO team now has capacity to work with you to land this.
